packages/next/src/build/webpack/loaders/next-edge-ssr-loader/index.ts

```diff
@@ -255,12 +257,22 @@ const edgeSSRLoader: webpack.LoaderDefinitionFunction<EdgeSSRLoaderQuery> =
 
     export const ComponentMod = pageMod
 
-    export default function(opts) {
-      return adapter({
+    export default async function(opts, event) {
+      const res = await adapter({
         ...opts,
         IncrementalCache,
         handler: render
       })
+
+      if (event && event.waitUntil) {
+        event.waitUntil(
+          Promise.all(
+            [...res?.waitUntil, ...globalThis.__next_private_global_wait_until__]
```

Comment:
```suggestion
            [res?.waitUntil, ...globalThis.__next_private_global_wait_until__]
```
Should we be spreading the original waitUntil since it should be a promise?